flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableInputFormat.java

```diff
@@ -207,19 +211,22 @@ public Row nextRecord(Row ignore) throws IOException {
 		if (reachedEnd()) {
 			return null;
 		}
-		Row row = new Row(rowArity);
+		Row row = new Row(fields.length);
```

Comment:
Why not use the row in `nextRecord(Row ignore)`? with reuse this row, you can never reassign partition values after initialization.